src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -177,6 +177,17 @@ type RedpandaConfig struct {
 	GroupTopicPartitions int `json:"groupTopicPartitions,omitempty"`
 	// Enable auto-creation of topics. Reference https://kafka.apache.org/documentation/#brokerconfigs_auto.create.topics.enable
 	AutoCreateTopics bool `json:"autoCreateTopics,omitempty"`
+	// For Redpanda configuration not exposed, a map can be provided for string values
+	// that are passed transparently to Redpanda.
+	// Notes:
+	// 1. versioning is not supported for the map keys
+	// 2. key names not supported by Redpanda will lead to failure on start up
+	// 3. updating this map requires a manual restart of the Redpanda pods
+	// Example input
+	// otherConfig:
+	//  enable_idempotence: "true"
+	//  default_topic_partitions: "3"
+	OtherConfig map[string]string `json:"otherConfig,omitempty"`
```

Comment:
This is copied to `RedpandaConfig.Other` which is a `map[string]interface{}`, would changing it allow nested JSON config such as:
```yaml
schema_registry.schema_registry_api: [{"name":"external","address":"redpanda","port":8081}]
```

Reply:
True, but the `interface{}` handling in `v1alpha1/zz_generated.deepcopy.go` cannot be auto-generated and I'd rather we don't have to adjust that one unless we really needed to.

We just need to add "other" in the rpk schema under each subsystem (with the new approach I'm following - see my update below).

Reply:
So this can only handle string values?

The config docs have this:
```yaml
default_topic_partitions: 1
```
Will this PR produce this in the config file:
```yaml
default_topic_partitions: "1"
```
Is that valid?

Reply:
It takes "1" but produces 1.

Reply:
```
pandaproxy_client:
  brokers:
  - address: one-node-cluster-0.one-node-cluster.default.svc.cluster.local.
    port: 9092
  retries: 123
```

Reply:
So does my arbitrary json example work?

Reply:
Yeah, it works but I had to pass it as a string (put quotes around it).

Here's `/etc/redpanda/redpanda.yaml` from the pod:

```yaml
...
  tune_disk_write_cache: false
  tune_fstrim: false
  tune_network: false
  tune_swappiness: false
  tune_transparent_hugepages: false
schema_registry:
  schema_registry_api:
    - address: redpanda
      name: external
      port: 8081
```

However, I think we should expose schema registry in the CRD. This is mostly meant for the 10s of single parameters we'd normally not want to configure (and for testing).

Reply:
Great news!

I agree we should expose it in the CRD, but it means that we can truly add arbitrary config, and I can try new features without modifying all the things first.